pkg/bootstrap/docker/openshift/helper.go

```diff
@@ -777,32 +779,34 @@ func (h *Helper) updateConfig(configDir string, opt *StartOptions) error {
 		certPath := filepath.Join(configDir, aggregatorCert)
 		keyPath := filepath.Join(configDir, aggregatorKey)
 
-		// TODO: reconcile this oadm logic with https://github.com/openshift/origin/blob/master/pkg/bootstrap/docker/openshift/admin.go#L121-L149
-		out, err := localcmd.New("oc").Args(
-			"adm",
-			"ca",
-			"create-signer-cert",
-			"--cert", cacertPath,
-			"--key", cakeyPath,
-			"--serial", caserialPath,
-		).CombinedOutput()
+		cmdOutput := &bytes.Buffer{}
+		createSignerCertOptions := &admin.CreateSignerCertOptions{
+			CertFile:   cacertPath,
+			KeyFile:    cakeyPath,
+			SerialFile: caserialPath,
+			Output:     cmdOutput,
+		}
+		_, err = createSignerCertOptions.CreateSignerCert()
 		if err != nil {
-			return errors.NewError(fmt.Sprintf("failed generating signer certificate, command output: %s\nerror: %v", out, err))
-		}
-
-		// TODO: reconcile this oadm logic with https://github.com/openshift/origin/blob/master/pkg/bootstrap/docker/openshift/admin.go#L121-L149
-		out, err = localcmd.New("oc").Args(
-			"adm",
-			"create-api-client-config",
-			"--certificate-authority", cacertPath,
-			"--signer-cert", cacertPath,
-			"--signer-key", cakeyPath,
-			"--signer-serial", caserialPath,
-			"--user", "aggregator-front-proxy",
-			"--client-dir", configDir,
-		).CombinedOutput()
+			return errors.NewError("cannot create signer cert").WithCause(err)
```

Comment:
"do as i say, not as i did", eh? https://github.com/openshift/origin/blob/master/pkg/bootstrap/docker/openshift/admin.go#L136-L157